toncli/src/binding/ts.rs

```diff
@@ -0,0 +1,237 @@
+                }
+                ts.push_str(ident);
+                ts.push_str("}");
+            }
+        }
+        Type::EnumOfTypes(variants) => {
+            let mut is_first = true;
+            for variant in variants {
+                if !is_first {
+                    ts.push_str(" | ");
+                }
+                ts.push_str(&format!("{{ {}: ", variant.name));
+                match &variant.value {
+                    Type::Struct(fields) if fields.len() == 1 && fields[0].name.is_empty() => {
+                        generate_type_decl(&fields[0].value, ident, ts)?;
+                    }
+                    _ => {
+                        generate_type_decl(&variant.value, ident, ts)?;
+                    }
+                }
+                ts.push_str(" }");
+                is_first = false;
+            }
+        }
+        Type::Array(item) => {
+            generate_type_decl(&item, ident, ts)?;
+            ts.push_str("[]");
+        }
+        Type::String { .. } => {
+            ts.push_str("string");
+        }
+        Type::Any { .. } => {
+            ts.push_str("any");
+        }
+        Type::BigInt { .. } => {
+            ts.push_str("bigint");
+        }
+        Type::Boolean { .. } => {
+            ts.push_str("boolean");
+        }
+        Type::EnumOfConsts(variants) => {
+            let mut is_first = true;
+            for variant in variants {
+                if !is_first {
+                    ts.push_str(" | ");
+                }
+                match &variant.value {
+                    ConstValue::Number(n) => ts.push_str(n.as_str()),
+                    ConstValue::String(s) => ts.push_str(&format!("\"{}\"", s)),
+                    ConstValue::Bool(b) => ts.push_str(b.as_str()),
+                    ConstValue::None {} => ts.push_str(&format!("\"{}\"", variant.name)),
+                }
+                is_first = false;
+            }
+        }
+        Type::Number {} => ts.push_str("number"),
+        Type::Generic { name, .. } => ts.push_str(&format!("{}<>", name)),
+    }
+    Ok(())
+}
+
+fn generate_type(_api: &API, _module: &Module, ty: &Field, ts: &mut String) -> CliResult<()> {
+    ts.push_str(&format!("export type {} = ", ty.name));
+    generate_type_decl(&ty.value, "", ts)?;
+    ts.push_str(";\n\n");
+    Ok(())
+}
+
+fn generate_function(
+    _api: &API,
+    module: &Module,
+    function: &Function,
+    ts: &mut String,
+) -> CliResult<()> {
+    ts.push_str(&format!("    {}(", camel(parse_snake(&function.name))));
+    let mut is_first = true;
+    for param in &function.params {
+        if !is_first {
+            ts.push_str(", ");
+        }
+        ts.push_str(&format!("\n        {}: ", param.name));
+        generate_type_decl(&param.value, "", ts)?;
+        is_first = false;
+    }
+    if !is_first {
+        ts.push_str(", ");
+    }
+    ts.push_str("\n        responseHandler?: ResponseHandler\n    ): Promise<");
```

Comment:
Include `responseHandler` parameter for only  functions using callback for events emiting

Reply:
binding gen ported to TS and moved to tools folder